astropy/timeseries/sampled.py

```diff
@@ -92,7 +92,7 @@ def __init__(self, data=None, *, time=None, time_start=None,
         elif time is not None and time_start is not None:
             raise TypeError("Cannot specify both 'time' and 'time_start'")
 
-        if time is not None and not isinstance(time, Time):
+        if time is not None and not isinstance(time, (Time, TimeDelta)):
```

Comment:
Would it make sense to just import `TimeBase` and check for that?

Reply:
I thought about this, but in the end I think that the code is more readable and explicit this way, as in it tells the non-expert reader immediately that the results apply to both absolute and delta times.

It's an arbitrary line, but I thought that code inside of the time subpackage should use `TimeBase` (internal) while the rest would be explicit. I guess there is some philosophy here -- if we or someone else adds a new subclass of `TimeBase`, do we want that to be in or out from consideration? One can make valid arguments both ways.

Reply:
@taldcroft - indeed, and the advantage of `Time, TimeDelta` was that both were imported already. Note that I changed it to `TimeBase` in the later commit, but that was really because I thought that might be the reason for the doctest failure (which was instead that the same change had to be made in `binned.py`).